earth2mip/initial_conditions/__init__.py

```diff
@@ -64,16 +89,27 @@ def get_initial_condition_for_model(
     # stack the history
     array = np.stack(arrays, axis=0)
 
-    # make an empty batch dim
-    array = array[None]
-
-    index = [data_source.channel_names.index(c) for c in time_loop.in_channel_names]
-    values = array[:, :, index]
-    regridder = regrid.get_regridder(data_source.grid, time_loop.grid).to(
-        time_loop.device
-    )
-    # TODO make the dtype flexible
-    x = torch.from_numpy(values).to(time_loop.device).type(torch.float)
+    index = [data_source.channel_names.index(c) for c in channel_names]
+    values = np.take(array, index, axis=1)
+    regridder = regrid.get_regridder(data_source.grid, grid).to(device)
+    x = torch.from_numpy(values).to(device).type(dtype)
     # need a batch dimension of length 1
+    # make an empty batch dim
+    x = x[None]
     x = regridder(x)
     return x
+
+
+def get_initial_condition_for_model(
+    time_loop: time_loop.TimeLoop, data_source: base.DataSource, time: datetime
+) -> torch.Tensor:
+    return get_data_from_source(
+        data_source,
+        time,
+        time_loop.in_channel_names,
+        time_loop.grid,
+        time_loop.n_history_levels,
+        time_loop.history_time_step,
+        time_loop.device,
+        time_loop.dtype,
+    )
```

Comment:
I added this indirection so that I didn't need to touch unrelated parts of the code that rely on `get_initial_condition_for_model` (e.g. ensemble inference). The examples also use `get_initial_condition_for_model`.

If we want to change the public APIs and remove this layer of indirection, we can do so in a separate PR.

Reply:
Indeed, we will wrap this in the discussion about data movement design.